spec.bs

```diff
@@ -13,3 +13,538 @@ Group: patcg
+      1. If |requestOrigin| is not a [=potentially trustworthy origin=], then return.
+      1. If |initiatorWindow| is not an [=environment settings object=], then return.
+      1. If |initiatorWindow| is not a [=secure context=], then return.
+      1. For each feature |f| in « "<code>browsing-topic</code>", "<code>interest-cohort</code>" »:
+          1. Run the <a href="https://www.w3.org/TR/permissions-policy-1/#algo-should-request-be-allowed-to-use-feature">Should request be allowed to use feature?</a> algorithm with <var ignore=''>feature</var> set to |f| and <var ignore=''>request</var> set to |request|. If the algorithm returns false, then return.
+
+          Note: the above algorithm should include the <a href="https://github.com/w3c/webappsec-permissions-policy/pull/499">pending update</a>, i.e. the |request| should be considered to contain the equivalent opt-in flags for both "browsing-topic" and the "interest-cohort" feature.
+      1. Let |topicsCallerContext| be a [=topics caller context=] with default initial field values.
+      1. Set |topicsCallerContext|'s [=topics caller context/caller origin=] to |requestOrigin|.
+      1. Set |topicsCallerContext|'s [=topics caller context/top level context domain=] to the result of running |initiatorWindow|'s [=environment settings object/global object=]'s [=Window/navigable=]'s [=navigable/top-level traversable=]'s [=navigable/active document=]'s {{Document/domain}} getter steps.
+      1. Let |unsafeMoment| be the [=wall clock=]'s [=wall clock/unsafe current time=].
+      1. Let |moment| be the result of running [=coarsen time=] algorithm given |unsafeMoment| and [=wall clock=] as input.
+      1. Let |now| be the [=duration from=] [=unix epoch=] to |moment|.
+      1. Set |topicsCallerContext|'s [=topics caller context/timestamp=] to |now|.
+      1. If the user preference setting disallows the access to topics from |topicsCallerContext|'s [=topics caller context/caller origin=] or |topicsCallerContext|'s [=topics caller context/top level context domain=], then return.
+      1. Let |topics| be the result from running the [=calculate the topics for caller=] algorithm, with |topicsCallerContext| as input.
+      1. Let |headerStructuredFields| be an empty Structured Fields <a href="https://www.rfc-editor.org/rfc/rfc8941.html#name-lists">List</a>.
+      1. For each |topic| in |topics|:
+          1. Let |topicItem| be a Structured Fields <a href="https://www.rfc-editor.org/rfc/rfc8941.html#name-integers">Integer</a> with value |topic|["{{BrowsingTopic/topic}}"].
+          1. Let |topicParameters| be an empty Structured Fields <a href="https://www.rfc-editor.org/rfc/rfc8941.html#name-parameters">Parameters</a>.
+          1. Set |topicParameters|["<code>version</code>"] to |topic|["{{BrowsingTopic/version}}"].
+          1. Set |topicParameters|["<code>config_version</code>"] to |topic|["{{BrowsingTopic/configVersion}}"].
```

Comment:
I am unsure about this snake_case capitalization. I opened an issue to see if we can get some clarity and documentation for the future: https://github.com/w3ctag/design-principles/issues/422

No need to change anything now, but something we should keep an eye on as the spec proceeds.